src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1736,7 +1736,7 @@ def generate_anchors(self, spatial_shapes=None, grid_size=0.05, device="cpu", dt
         anchors = torch.concat(anchors, 1)
         valid_mask = ((anchors > eps) * (anchors < 1 - eps)).all(-1, keepdim=True)
         anchors = torch.log(anchors / (1 - anchors))
-        anchors = torch.where(valid_mask, anchors, torch.inf)
+        anchors = torch.where(valid_mask, anchors, torch.tensor(float("inf"), dtype=dtype, device=device))
```

Comment:
```suggestion
         
        anchors = torch.where(valid_mask, anchors, torch.finfo(dtype).min, dtype=dtype, device=device))
```
if the dtype is the dtype of `valid_mask` then this would make more sense ! Otherwise the min (float32 -inf) is not gonna be the same!

Reply:
dtype is the dtype of `anchors`, `valid_mask` is a condition, dtypes are the same but just by construction. 

But you're right, `torch.finfo(dtype).max` is good :)

Reply:
6b0f18a32d0b46bc56489e8d47621c350e52a038